source/server/modes.cpp

```diff
@@ -0,0 +1,291 @@
+/*   modes.cpp                                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: fbarros <fbarros@student.42.fr>            +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/08/09 19:22:13 by fbarros           #+#    #+#             */
+/*   Updated: 2023/09/09 15:07:55 by fbarros          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <Server.hpp>
+#include <cstdlib>
+
+void Server::replyMode(const Client& client, const std::string& channel, const std::string& param1, const std::string& param2)
+{
+    const std::string nick(client.getNickname());
+
+    // "Official" reply
+    if (param1 == "+l")
+        broadcast(":" + client.toString() + " MODE #" + channel + " " + param1 + " :" + param2 + EOL, channel);
+    else
+        broadcast(":" + client.toString() + " MODE #" + channel + " :" + param1 + EOL, channel);
+}
+
+std::string getNotChanopMsg(const std::string& token)
+{
+    if (token == "+i")
+    {
+        return "You must be channel operator to set channel mode i (invite only)";
+    }
+    else if (token == "-i")
+    {
+        return "You must be channel operator to unset channel mode i (invite only)";
+    }
+    else if (token == "+t")
+    {
+        return "You must be channel operator to set channel mode t (topic restriction)";
+    }
+    else if (token == "-t")
+    {
+        return "You must be channel operator to unset channel mode t (topic restriction)";
+    }
+    else if (token == "+k")
+    {
+        return "You must be channel operator to set channel mode k (key)";
+    }
+    else if (token == "-k")
+    {
+        return "You must be channel operator to unset channel mode k (key)";
+    }
+    else if (token == "+o")
+    {
+        return "You must be channel operator to set channel mode o (channel operator)";
+    }
+    else if (token == "-o")
+    {
+        return "You must be channel operator to unset channel mode o (channel operator)";
+    }
+    else if (token == "+l")
+    {
+        return "You must be channel operator to set channel mode l (limit)";
+    }
+    else if (token == "-l")
+    {
+        return "You must be channel operator to unset channel mode l (limit)";
+    }
+    return "";
+}
+
+void Server::execMODE(Client& client, const std::string line)
+{
+    // TODO: might need to check for other symbols ('&','+','!')
+    const int id(client.getId());
+    std::istringstream iss(line);
+    std::string channelName;
+    std::queue<std::string> modes;
+    std::string args;
+
+    iss >> channelName;
+    if (channelName.empty())
+    {
+        addMessage("MODE :<target> [[(+|-)]<modes> [<mode-parameters>]]\r\n", id);
```

Comment:
choose between addmessage or replymessage, you can set your reply message at the end of your command file